common/hello.go

```diff
@@ -0,0 +1,193 @@
+			err = managementClusterKubeClient.Delete(ctx, nginxConfigMap)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = managementClusterKubeClient.Delete(ctx, helloApp)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = managementClusterKubeClient.Delete(ctx, helloConfigMap)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+	})
+}
+
+// deployApp creates an `App` CR for the desired application.
+// It currently has some needed workarounds until we improve our `clustertest` framework.
+func deployApp(ctx context.Context, name, namespace string, organization *organization.Org, version, valuesFile string, values map[string]string) (*v1alpha1.App, *v1.ConfigMap) {
+	var err error
+	managementClusterKubeClient := state.GetFramework().MC()
+	appBuilder := application.New(fmt.Sprintf("%s-%s", state.GetCluster().Name, name), name).
+		WithCatalog("giantswarm").
+		WithOrganization(*organization).
+		// If we don't pass any version value when building the app, the latest version will be calculated and used. But the calculation will fail when the app name doesn't match the repository name.
+		WithVersion(version).
+		WithInCluster(false).
+		// We need to manually set this label that should be automatically added once we can set the `app.Config` property.
+		WithAppLabels(map[string]string{"giantswarm.io/cluster": state.GetCluster().Name})
+
+	if valuesFile != "" {
+		appBuilder, err = appBuilder.WithValuesFile(path.Clean(valuesFile), &application.TemplateValues{
+			ClusterName:  state.GetCluster().Name,
+			Organization: state.GetCluster().Organization.Name,
+			ExtraValues:  values,
+		})
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+
+	app, configMap, err := appBuilder.Build()
+	Expect(err).ShouldNot(HaveOccurred())
+
+	err = managementClusterKubeClient.Create(ctx, configMap)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// We need to set these properties manually after building, because they are calculated using the `app.Config` property that we currently can't set.
+	app.Spec.Config.ConfigMap.Name = fmt.Sprintf("%s-cluster-values", state.GetCluster().Name)
+	app.Spec.Config.ConfigMap.Namespace = organization.GetNamespace()
+	app.Spec.KubeConfig.Context.Name = fmt.Sprintf("%s-admin@%s", state.GetCluster().Name, state.GetCluster().Name)
+	app.Spec.KubeConfig.Secret.Name = fmt.Sprintf("%s-kubeconfig", state.GetCluster().Name)
+
+	// We need to set the namespace that will be used when installing the chart in the WC. It's not the organization namespace, because we don't have orgs in WCs.
+	app.Spec.Namespace = namespace
+	err = managementClusterKubeClient.Create(ctx, app)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	return app, configMap
+}
+
+func getWorkloadClusterDnsZone(ctx context.Context) string {
+	clusterValuesConfigMap := &v1.ConfigMap{}
+	err := state.GetFramework().MC().Get(ctx, ctrl.ObjectKey{Namespace: state.GetCluster().Organization.GetNamespace(), Name: fmt.Sprintf("%s-cluster-values", state.GetCluster().Name)}, clusterValuesConfigMap)
```

Comment:
Why aren't we using `GetHelmValues` here? 

E.g. 

```go
values = &application.DefaultAppsValues{}
err := state.GetFramework().MC().GetHelmValues(state.GetCluster().Name, state.GetCluster().GetNamespace(), values)
Expect(err).NotTo(HaveOccurred())

if values.BaseDomain == "" {
	Fail("baseDomain field missing from cluster-values configmap")
}

return values.BaseDomain
```

Reply:
`GetHelmValues` calls `GetReleaseValues` on the `helmClient` so should return back the final values used with the chart after all the config layering and defaulting.

Reply:
Changed it to use `GetHelmValues`